pkg/registry/pod/etcd/etcd.go

```diff
@@ -218,6 +220,38 @@ type LogREST struct {
 	kubeletConn client.ConnectionInfoGetter
 }
 
+// Max length read from the body of a log location which returns error status
+const (
+	maxReadLength = 1000
+)
+
+// A log location response checker to transform the error.
+type LogLocationResponseChecker struct {
+	podName string
```

Comment:
But looking at the logic of the checker, everything is standard and applicable to basically any HTTP call. If we mutate the kubelet api, we can always copy this back.